peer-prep/src/DataModel/QuestionModel.js

```diff
@@ -0,0 +1,69 @@
+// Class that stores and encapsulates the Question and its related fields
+
+class QuestionModel {
+  static QUESTION_FIELDS = {
+    ID: "Id",
+    TITLE: "Title",
+    DESCRIPTION: "Description",
+    COMPLEXITY: "Complexity",
+    CATEGORY: "Category",
+  };
+
+  static ERROR_MESSAGE = " Is Empty";
+
+  constructor(id, title, description, complexity, category) {
+    this.id = id;
+    this.title = title;
+    this.description = description;
+    this.complexity = complexity;
+    this.category = category;
+  }
+
+  isQuestionValid() {
+    let errors = [];
+
+    for (const qField in QuestionModel.QUESTION_FIELDS) {
+      const fieldName = QuestionModel.QUESTION_FIELDS[qField].toLowerCase();
+
+      if (
+        (typeof this[fieldName] === "string" &&
+          this[fieldName].trim() === "") ||
+        !this[fieldName]
```

Comment:
Glad to see that you've implemented a validity function for checking the input to the QuestionModel!